miden/src/cli/data.rs

```diff
@@ -80,33 +103,56 @@ impl OutputFile {
         Ok(outputs)
     }
 
-    /// write the output file
+    /// Write the output file
     pub fn write(outputs: Vec<u64>, path: &Option<PathBuf>) -> Result<(), String> {
-        match path {
-            Some(path) => {
-                // if path provided create output file
-
-                println!("Creating output file `{}`", path.display());
-
-                let file = fs::File::create(&path).map_err(|err| {
-                    format!(
-                        "Failed to create output file `{}` - {}",
-                        path.display(),
-                        err
-                    )
-                })?;
-
-                println!("Writing data to output file");
-
-                // write outputs to output file
-                serde_json::to_writer_pretty(file, &Self { outputs })
-            }
-
-            // no path provided - write outputs to stdout
-            None => serde_json::to_writer_pretty(std::io::stdout(), &Self { outputs }),
+        if let Some(path) = path {
+            // if path provided, create output file
+            println!("Creating output file `{}`", path.display());
+
+            let file = fs::File::create(&path).map_err(|err| {
+                format!(
+                    "Failed to create output file `{}` - {}",
+                    path.display(),
+                    err
+                )
+            })?;
+
+            println!("Writing data to output file");
+
+            // write outputs to output file
+            serde_json::to_writer_pretty(
+                file,
+                &Self {
+                    outputs: outputs
+                        .iter()
+                        .map(|v| v.to_string())
+                        .collect::<Vec<String>>(),
+                },
```

Comment:
Let's create a constructor which would handle conversion from integers to strings. Something like:
```Rust
pub fn new(data: Vec<u64>) -> Self;
```